source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp

```diff
@@ -0,0 +1,429 @@
+            filterProcessingDurations.push_back(filterProcessingDurationMedian);
+            filterProcessingDurations.push_back(filterProcessingDurationMean);
+        }
+        printFilterResults(filterProcessingDurations);
+    }
+    void benchmarkCapture2D(Zivid::Camera &camera, const size_t &exposureTime, const size_t &numFrames)
+    {
+        printCapture2DHeader(numFrames, exposureTime);
+
+        Zivid::Settings2D settings;
+        settings.set(Zivid::Settings2D::ExposureTime(std::chrono::microseconds{ exposureTime }));
+
+        camera.capture2D(settings);
+
+        std::vector<Duration> captureDurations;
+        std::vector<std::vector<Duration>> durations;
+        for(size_t i = 0; i < numFrames; i++)
+        {
+            const auto beforeCapture = HighResClock::now();
+            const auto frame2D = camera.capture2D(settings);
+            const auto afterCapture = HighResClock::now();
+
+            captureDurations.push_back(afterCapture - beforeCapture);
+        }
+        durations.push_back(captureDurations);
+        printCapture2DResults(durations);
+    }
+    void benchmarkSave(Zivid::Camera &camera, const size_t &numFrames)
+    {
+        printSaveHeader(numFrames);
+
+        auto frame = camera.capture();
+        frame.getPointCloud();
+
+        std::vector<std::vector<Duration>> saveDurations;
+        std::vector<std::string> fileNames = { "Zivid3D.zdf", "Zivid3D.ply", "Zivid3D.pcd", "Zivid3D.xyz" };
+        for(auto fileName : fileNames)
+        {
+            std::vector<Duration> durationsPerFormat;
+            for(size_t j = 0; j < numFrames; j++)
+            {
+                const auto beforeSave = HighResClock::now();
+                frame.save(fileName);
+                const auto afterSave = HighResClock::now();
+
+                durationsPerFormat.push_back(afterSave - beforeSave);
+            }
+            saveDurations.push_back(durationsPerFormat);
+        }
+        printSaveResults(saveDurations);
+    }
+} // namespace
+
+int main()
+{
+    try
+    {
+        Zivid::Application zivid;
+
+        printZividInfo(zivid);
+
+        printPrimarySeparationLine();
+        std::cout << "             Starting Zivid Benchmark              " << std::endl;
+
+        const size_t numConnects = 10;
+        benchmarkConnect(zivid, numConnects);
+
+        auto camera = zivid.connectCamera();
+
+        size_t exposureTime = 6500;
+        if(camera.modelName().substr(0, 14) != "Zivid One Plus") exposureTime = 8333;
+
+        const size_t numFrames3D = 20;
+        const std::vector<unsigned int> irisOneFrame = { 21U };
+        const std::vector<size_t> exposureTimeOneFrame = { exposureTime };
+        std::vector<Zivid::Settings> settingsOneFrame(irisOneFrame.size());
+
+        std::vector<std::vector<Duration>> captureDurationsOneFrame;
+
+        setSettingsVector(settingsOneFrame, irisOneFrame, exposureTimeOneFrame, false, false);
```

Comment:
The code from here and down is a wall of text, some creating some functsion should solve that.

Also all looks very similar, maybe the same function proposed above can be reused or that you can make some loops here?

Reply:
I don't understand what you mean here.

Reply:
**Wall of text**
So much variables, functions calls and so on going on [here](https://github.com/zivid/cpp-extra-samples/blob/3c2cebaa51aaa2a3a6ba50789060571b856247e3/source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp#L402-L455), I don't even know where to start looking at it. Nothink that looks like logical blocks.

**Functions**
You can add structure to this by factoring this out in probably 2-3 functions and only call them here. For example [ZividBenchmark.cpp#L429-L455](https://github.com/zivid/cpp-extra-samples/blob/3c2cebaa51aaa2a3a6ba50789060571b856247e3/source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp#L429-L455) looks like something that could be a function.

**Reuse**
Once you have factored out 2-4 function, you may see that some of them are almost similar (maybe not). Then, maybe you can delete one of them and just by passing an an extra parameter to the function remaining function to achieve the same.

Reply:
Done.